### PR TITLE
feat/fix: reactive filename/ fallback for __file__

### PR DESCRIPTION
**Pull request by dmadisetti**
Followup to the tests, seemed like a quick change when reading over #1989 #1988

@akshayka

### CODE REVIEW

**Review by akshayka — DISMISSED** [review]
Ty again!

**Comment by dmadisetti** [comment]
tmpdirs are too long on Mac: https://github.com/pytest-dev/pytest/issues/5802

**Comment by akshayka** [comment]
> tmpdirs are too long on Mac: https://github.com/pytest-dev/pytest/issues/5802

I'm fine with skipping the test on Darwin

**Comment by dmadisetti** [comment]
Failure on windows made me realize that variable output is truncated to 50 chars. Still a quick fix

File: marimo/_runtime/runtime.py [hunk 1]
```diff
@@ -1116,6 +1117,31 @@ async def run(
...
+            execution_type=self.execution_type,
+            execution_context=self._install_execution_context,
+            preparation_hooks=self._preparation_hooks,
+            pre_execution_hooks=self._pre_execution_hooks,
+            post_execution_hooks=self._post_execution_hooks,
+            on_finish_hooks=(
+                self._on_finish_hooks + [self._broadcast_missing_packages]
+            ),
+        )
+        await runner.run_all()
```
Comment (akshayka) [inline marimo/_runtime/runtime.py:1143]: I missed this in my first review. This should be:

```
await self._run_cells(roots)
```

which wraps the `Runner` and does some additional housekeeping, including handling `mo.state` updates.

Comment (dmadisetti) [reply marimo/_runtime/runtime.py]: Much cleaner

**Review by akshayka — APPROVED** [review]

**Pull request merged**